core/templates/dev/head/forms/formBuilder.js

```diff
@@ -1274,11 +1274,61 @@ oppia.directive('schemaBasedUnicodeEditor', [function() {
     },
     templateUrl: 'schemaBasedEditor/unicode',
     restrict: 'E',
-    controller: ['$scope', '$filter', '$sce', 'parameterSpecsService',
-        function($scope, $filter, $sce, parameterSpecsService) {
+    controller: ['$scope', '$filter', '$sce', '$window', 'parameterSpecsService',
+        function($scope, $filter, $sce, $window, parameterSpecsService) {
       $scope.allowedParameterNames = parameterSpecsService.getAllParamsOfType('unicode');
       $scope.doUnicodeParamsExist = ($scope.allowedParameterNames.length > 0);
 
+      $scope.allowSpeechRecognition = (
+        $scope.uiConfig() &&
+        $scope.uiConfig().speechRecognitionLanguage &&
+        !!$window.webkitSpeechRecognition);
+      $scope.isCurrentlyRecording = false;
+
+      var recognition = new webkitSpeechRecognition();
+
+      $scope.toggleSpeechRecognition = function() {
```

Comment:
[From @jacobdavis11] I'm afraid this is confusing to me. What exactly is happening when you toggleSpeechRecognition to off?

I would have expected there to be two different functions, one to activate speech recognition and one to disactivate it.